### DISCUSSION TITLE
Constraints for array variable on boundaries

### DISCUSSION THREAD

**Discussion by pj131611** [idaholab/moose#28860] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Greetings!

I'm looking for a constraint function that constrain the boundary value of the same variable through multiple boundaries. But the problem is I'm working with array variables and I couldn't find a constraint for array. So my question is does such function exist? If not, would it be reasonable to write a constraint function that inherit from ```EqualValueBoundaryConstraint``` or ```TiedValueConstraint``` but with array variable and component as input?

Thanks

**Comment by GiudGiud** [comment]
Hello

No I don't think we have that at the moment but feel free to implement it. 
Please let us know if you hit any difficulties doing so

**Reply by pj131611** [reply]
Cool. I started with inherit from Constraint class and replace the variable with arraymoosevariable and here comes the first problem while testing:

```
*** ERROR ***
Unknown type of Constraint object
```

After doing a little bit of digging around, it seems like I need add this new constraint into Constraint Warehouse and objects, which will make me re-make moose. Is there a way to avoid this? It takes way too long to re-make the whole moose framework.

**Reply by GiudGiud** [reply]
hello

the Constraint class should be the base class for your new objects most likely.
It only brings a `variable` parameter which you can use and some background calls to the subproblem to trigger the right kind of re-iniit. 

If you inherit from the Constraint class, your object will be added to the Constraint warehouse as expected

**Reply by pj131611** [reply]
Hi there! 

A surprising problem arise while I'm trying to simplify the mesh: Is there a way to convert the input unstructured mesh(triangles and simplex) to structured mesh(squares and cubes)? It seems a lot easier to play with structured mesh.

**Reply by GiudGiud** [reply]
not in a general way. Most shapes described with unstructured mesh cannot be described with structured meshes

You can generate simple structured meshes with this:
https://mooseframework.inl.gov/source/meshgenerators/CartesianMeshGenerator.html

Under the hood they are still represented as unstructured meshes though. MOOSE does not support the libmesh structured mesh
